pgjdbc/src/main/java/org/postgresql/jdbc/EscapedFunctions.java

```diff
@@ -683,64 +658,86 @@ private static String constantToDatePart(String type) throws SQLException {
   /**
    * user translation
    *
    * @param parsedArgs arguments
-   * @return sql call
    * @throws SQLException if something wrong happens
    */
-  public static String sqluser(List<?> parsedArgs) throws SQLException {
+  public static void sqluser(StringBuilder buf, List<CharSequence> parsedArgs) throws SQLException {
+    zeroArgumentFunctionCall(buf, "user", "user", parsedArgs);
+  }
+
+  private static void zeroArgumentFunctionCall(StringBuilder buf, String call, String functionName,
+      List<CharSequence> parsedArgs) throws PSQLException {
     if (!parsedArgs.isEmpty()) {
-      throw new PSQLException(GT.tr("{0} function doesn''t take any argument.", "user"),
+      throw new PSQLException(GT.tr("{0} function doesn''t take any argument.", functionName),
           PSQLState.SYNTAX_ERROR);
     }
-    return "user";
+    buf.append(call);
   }
 
-  private static String singleArgumentFunctionCall(String call, String functionName,
-      List<?> parsedArgs) throws PSQLException {
+  private static void singleArgumentFunctionCall(StringBuilder buf, String call, String functionName,
+      List<CharSequence> parsedArgs) throws PSQLException {
     if (parsedArgs.size() != 1) {
       throw new PSQLException(GT.tr("{0} function takes one and only one argument.", functionName),
           PSQLState.SYNTAX_ERROR);
     }
-    StringBuilder buf = new StringBuilder();
-    buf.append(call);
-    buf.append(parsedArgs.get(0));
-    return buf.append(')').toString();
+    CharSequence arg0 = parsedArgs.get(0);
+    buf.ensureCapacity(buf.length() + call.length() + arg0.length() + 1);
+    buf.append(call).append(arg0).append(')');
   }
 
-  private static String twoArgumentsFunctionCall(String call, String functionName,
-      List<?> parsedArgs) throws PSQLException {
+  private static void twoArgumentsFunctionCall(StringBuilder buf, String call, String functionName,
+      List<CharSequence> parsedArgs) throws PSQLException {
     if (parsedArgs.size() != 2) {
       throw new PSQLException(GT.tr("{0} function takes two and only two arguments.", functionName),
           PSQLState.SYNTAX_ERROR);
     }
-    StringBuilder buf = new StringBuilder();
-    buf.append(call);
-    buf.append(parsedArgs.get(0)).append(',').append(parsedArgs.get(1));
-    return buf.append(')').toString();
+    appendCall(buf, call, ",", ")", parsedArgs);
+  }
+
+  /**
+   * Appends {@code begin arg0 separator arg1 separator end} sequence to the input {@link StringBuilder}
+   * @param sb destination StringBuilder
+   * @param begin begin string
+   * @param separator separator string
+   * @param end end string
+   * @param args arguments
+   */
+  public static void appendCall(StringBuilder sb, String begin, String separator,
+      String end, List<CharSequence> args) {
+    int size = begin.length();
+    // Avoid Iterator instantiations just in case, so plain for, not forach
+    for (int i = 0; i < args.size(); i++) {
+      size += args.get(i).length();
```

Comment:
is this degree of precision necessary? can we approximate a guess (say 32) for average length of args instead?

Reply:
Why approximate when we have exact value?

Reply:
to avoid the work to calculate the exact value

Reply:
The strings would HAVE to be appended, so there will be a walk over the list.
Additional walk is likely to be just free.

Proper size estimation reduces allocation waste.